storybookjs/storybook
# Bug: version switcher of compositions

Closes https://github.com/storybookjs/storybook/issues/21505

## What I did

- I ensured the version switcher closes when you click outside
- I ensure the version switcher closes when you click another version
- I ensured the ref enters a loading state when a new version is being requested

## How to test

- add a ref

```
  refs: {
    'design-system': {
      title: 'Design System',
      url: 'https://master--5ccbc373887ca40020446347.chromatic.com',
    },
  },
```

... to `code/ui/.storybook/main.ts`

- start the ui storybook
- open the browser
- go to the ref
- change version

Expect that the version switcher closes
Expect that the ref enters the loading state
Expect that after some time, the ref shows again, and the same story is still selected. (assuming the same sotry existed in the other version)
